opp_path_selection/src/path_selection_artist.cpp

```diff
@@ -0,0 +1,515 @@
+  auto points_it = std::find_if(marker_array_.markers.begin(),
+                                marker_array_.markers.end(),
+                                [](const visualization_msgs::Marker& marker) { return marker.id == 0; });
+
+  // Convert the selection points to Eigen vectors
+  std::vector<Eigen::Vector3d> points;
+  for (const geometry_msgs::Point& pt : points_it->points)
+  {
+    Eigen::Vector3d e;
+    tf::pointMsgToEigen(pt, e);
+    points.push_back(std::move(e));
+  }
+
+  if (points.size() < 2)
+  {
+    res.cloud_indices.clear();
+    res.success = false;
+    res.message = "No Points Selected, use Geometric Axis";
+    return true;
+  }
+
+  // get point cloud from service request
+  pcl::PointCloud<pcl::PointXYZ>::Ptr cloud(new pcl::PointCloud<pcl::PointXYZ>());
+  pcl::fromROSMsg(req.input_cloud, *cloud);
+
+  // get parameters from file. Note: This file is specified on construction, re-read allows dynamic changes
+  PathSelectorParameters params;
+  bool success = message_serialization::deserialize(path_selection_config_file, params);
+  if (!success)
+  {
+    ROS_ERROR_STREAM("Could not load path selection config from: " << path_selection_config_file);
+    return false;
+  }
+
+  // get all indices from provided mesh vertices along path
+  // TODO replace getRegionOfInterest with a path equivalent
+  PathSelector sel;
+  res.cloud_indices = sel.findPointsAlongSegments(cloud, points, params);
+
+  if (!res.cloud_indices.empty())
+  {
+    res.success = true;
+    res.message = "Selection complete";
+  }
+  else
+  {
+    res.success = false;
+    res.message = "Unable to identify points along selected path";
+  }
+
+  return true;
+}
+
+bool PathSelectionArtist::collectPathPointsMeshCb(opp_msgs::GetPathSelectionMeshRequest& req,
+                                                  opp_msgs::GetPathSelectionMeshResponse& res)
+{
+  // get points from existing marker_array_
+  auto points_it = std::find_if(marker_array_.markers.begin(),
+                                marker_array_.markers.end(),
+                                [](const visualization_msgs::Marker& marker) { return marker.id == 0; });
+
+  // Convert the selection points to Eigen vectors
+  std::vector<Eigen::Vector3d> points;
+  for (const geometry_msgs::Point& pt : points_it->points)
+  {
+    Eigen::Vector3d e;
+    tf::pointMsgToEigen(pt, e);
+    points.push_back(std::move(e));
+  }
+
+  if (points.size() < 2)
+  {
+    res.mesh_indices.clear();
+    res.success = true;
+    res.message = "No Points Selected, use Geometric Axis";
+    return true;
+  }
+
+  // get parameters from file. Note: This file is specified on construction, re-read allows dynamic changes
+  PathSelectorParameters params;
+  bool success = message_serialization::deserialize(path_selection_config_file, params);
+  if (!success)
+  {
+    ROS_ERROR_STREAM("Could not load path selection config from: " << path_selection_config_file);
+    return false;
+  }
```

Comment:
Same comment about supporting dynamic behavior through the service definition

Reply:
same response, params not used, no longer read, not dynamic